### DISCUSSION TITLE
video mask in deforum ext. doesn't work

### DISCUSSION THREAD

**Discussion by mindmaze23** [community/community#47017] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hi
Trying to use video mask in deforum extention.

1. I copy the path
2. enbale use_video_mask
3. generate
4. an input frame and a input mask frame is created and filled with png's
5. The end result exclude and mask effect

thx
O

**Comment by numz** [comment]
Hi,
while waiting a real fix, perhaps you can try this.

in the file "stable-diffusion-webui\extensions\deforum-for-automatic1111-webui\scripts\deforum_helpers\render.py" in line 376, I replace :
```python
if args.use_mask:
    if not using_vid_init:
        args.mask_image = compose_mask_with_check(root, args, mask_seq, mask_vals, args.init_sample) if args.init_sample is not None else None # we need it only after the first frame anyway
```

by this : 
```python
if args.use_mask:
    if not using_vid_init:
        args.mask_image = compose_mask_with_check(root, args, mask_seq, mask_vals, args.init_sample) if args.init_sample is not None else None # we need it only after the first frame anyway
    else:
        image_frame, _ = load_img(args.init_image, shape=(args.W, args.H), use_alpha_as_mask=args.use_alpha_as_mask)
        args.mask_image = compose_mask_with_check(root, args, mask_seq, mask_vals, np.array(image_frame))
```
I also have to put a black image in the "init>mask init>mask file" parameter otherwise it combines mask with the default image.

it works for me

regards
